relayer/listener.go

```diff
@@ -233,141 +233,141 @@ func (r *Relayer) ProcessLogs(ctx context.Context) error {
 // 4. A match on sourceBlockchainID and destinationBlockchainID, with any originSenderAddress and any destinationAddress
-func (r *Relayer) getMessageRelayer(
+func (lstnr *Listener) getApplicationRelayer(
 	sourceBlockchainID ids.ID,
 	destinationBlockchainID ids.ID,
 	originSenderAddress common.Address,
 	destinationAddress common.Address,
-) (*messageRelayer, bool) {
+) (*applicationRelayer, bool) {
 	// Check for an exact match
-	messageRelayerID := database.CalculateRelayerID(
+	applicationRelayerID := database.CalculateRelayerID(
 		sourceBlockchainID,
 		destinationBlockchainID,
 		originSenderAddress,
 		destinationAddress,
 	)
-	if messageRelayer, ok := r.messageRelayers[messageRelayerID]; ok {
-		return messageRelayer, ok
+	if applicationRelayer, ok := lstnr.applicationRelayers[applicationRelayerID]; ok {
+		return applicationRelayer, ok
 	}
 
 	// Check for a match on sourceBlockchainID and destinationBlockchainID, with a specific originSenderAddress and any destinationAddress
-	messageRelayerID = database.CalculateRelayerID(
+	applicationRelayerID = database.CalculateRelayerID(
 		sourceBlockchainID,
 		destinationBlockchainID,
 		originSenderAddress,
 		database.AllAllowedAddress,
 	)
-	if messageRelayer, ok := r.messageRelayers[messageRelayerID]; ok {
-		return messageRelayer, ok
+	if applicationRelayer, ok := lstnr.applicationRelayers[applicationRelayerID]; ok {
+		return applicationRelayer, ok
 	}
 
 	// Check for a match on sourceBlockchainID and destinationBlockchainID, with any originSenderAddress and a specific destinationAddress
-	messageRelayerID = database.CalculateRelayerID(
+	applicationRelayerID = database.CalculateRelayerID(
 		sourceBlockchainID,
 		destinationBlockchainID,
 		database.AllAllowedAddress,
 		destinationAddress,
 	)
-	if messageRelayer, ok := r.messageRelayers[messageRelayerID]; ok {
-		return messageRelayer, ok
+	if applicationRelayer, ok := lstnr.applicationRelayers[applicationRelayerID]; ok {
+		return applicationRelayer, ok
 	}
 
 	// Check for a match on sourceBlockchainID and destinationBlockchainID, with any originSenderAddress and any destinationAddress
-	messageRelayerID = database.CalculateRelayerID(
+	applicationRelayerID = database.CalculateRelayerID(
 		sourceBlockchainID,
 		destinationBlockchainID,
 		database.AllAllowedAddress,
 		database.AllAllowedAddress,
 	)
-	messageRelayer, ok := r.messageRelayers[messageRelayerID]
-	return messageRelayer, ok
+	applicationRelayer, ok := lstnr.applicationRelayers[applicationRelayerID]
+	return applicationRelayer, ok
 }
 
 // RelayMessage relays a single warp message to the destination chain. Warp message relay requests from the same origin chain are processed serially
-func (r *Relayer) RelayMessage(warpLogInfo *vmtypes.WarpLogInfo, storeProcessedHeight bool) error {
-	r.logger.Info(
+func (lstnr *Listener) RelayMessage(warpLogInfo *vmtypes.WarpLogInfo, storeProcessedHeight bool) error {
```

Comment:
I find it a little odd that the `Listener` has a `RelayMessage` function (as opposed to that being contained in `ApplicationRelayer`. I don't think we should change anything functionally, but this function is really:
1. Filtering unsupported messages that won't be relayed.
2. Routing supported messages to the proper `ApplicationRelayer` to be relayed.

Any ideas for what we could rename it maybe?

Reply:
How about `RouteMessage`?